src/compute-client/src/protocol/command.rs

```diff
@@ -585,6 +619,10 @@ impl PeekTarget {
 pub struct Peek<T = mz_repr::Timestamp> {
     /// Target-specific metadata.
     pub target: PeekTarget,
+    /// The relation description for the rows returned by this peek, before
+    /// applying the [RowSetFinishing] but _after_ applying the given
+    /// `map_filter_project`.
+    pub result_desc: RelationDesc,
```

Comment:
I was wondering how `RowSetFinishing` can change the relation type. It is because it includes a projection, so it can remove columns, right? Do you know why we need that and can't do all the projections in the MFP instead?

Reply:
I don't know all the cases where this is needed, but I think one of them is `INSERT INTO ... SELECT ...` with a `RETURNING` clause. Where we do the peek on workers, then the insert in envd, and we can then still project out fields before returning rows.

Reply:
Ah, seems a bit strange that the `RETURNING` projection needs to be part of the peek description, rather than some post-processing that occurs after the peek. But I'm sure there is context I'm missing and it doesn't matter for this PR anyway.